packages/protocol/src/protocol.yml

```diff
@@ -647,6 +669,58 @@ LocalUtils:
       parameters:
         stacksId: string
 
+    newMockingProxy:
+      returns:
+        mockingProxy: MockingProxy
+
+MockingProxy:
+  type: interface
+
+  extends: EventTarget
+
+  initializer:
+    port: number
+    requestContext: APIRequestContext
+
+  commands:
+    setInterceptionPatterns:
+      parameters:
+        patterns:
+          type: array
+          items:
+            type: object
+            properties:
+              glob: string?
+              regexSource: string?
+              regexFlags: string?
+
+  events:
+    route:
+      parameters:
+        route: Route
+
+    request:
+      parameters:
+        request: Request
+        correlation: string
+
+    requestFailed:
+      parameters:
+        request: Request
+        failureText: string?
+        responseEndTiming: number
+
+    requestFinished:
+      parameters:
+        request: Request
+        response: Response?
```

Comment:
Why is there a response here? Shouldn't there be a `response` event beforehand?

Reply:
I pasted this from `BrowserContext#requestFinished`. I think it's here to save on roundtrip. Without it, this would need to `await request.response()` first:

https://github.com/microsoft/playwright/blob/cd7f3b6e65d25defdbf27da5d133cff586ef7d17/packages/playwright-core/src/client/browserContext.ts#L196-L197